prow/docs/pr_status_setup.md

```diff
@@ -0,0 +1,57 @@
+# How to setup PR Status 
+This document helps configure PR Status endpoints. 
+
+## Setup secrets
+PR status is an OAuth App that query pull requests on behalf of the authenticated users.
+Therefore, some secret pieces of information are needed to authorize users for the app. The following
+steps will show you how to setup an oauth app that works with PR Status.
+1. Create your Github Oauth application 
+
+    https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/
+    
+    The callback url should be:
+    
+    `<protocol>://<doman_name>:<port>/github-redirect`
+2. Create a secret file for github oauth that content should be as following. The information can be found in the [Github OAuth developer settings](https://github.com/settings/developers):
+    
+    ```
+    client_id: <APP_CLIENT_ID>
+    client_secret: <APP_CLIENT_SECRET>
+    redirect_url: <protocall>://<doman_name>:<port>/github-redirect
+    final_redirect_url: <protocol>://<doman_name>:<port>/pr
+    ```
+3. Create another secret file for the cookie store. The file should contain a random 64-bytes key
+4. Use `kubectl`, which should already point to your Prow cluster, to create secrets using the command:
+    
+    `kubectl create secret generic <GITHUB_OAUTH_SECRET_NAME> --from-file=<PATH_TO_YOUR_GITHUB_SECRET>`
+    
+    `kubectl create secret generic <COOKIE_SECRET_NAME> --from-file=<PATH_TO_YOUR_COOKIE_KEY_SECRET>`
+5. Open `test-infra/prow/cluster/deck_deployment.yaml`. [Mount](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets) secrets to your deck volume:
+    
+    Under `volumes` token, add:
+    ```
+    - name: <GITHUB_OAUTH_SECRET_VOLUME_NAME>
+      secret:
+          secretName: <GITHUB_OAUTH_SECRET_NAME>
+    - name: <COOKIE_SECRET_VOLUME_NAME>	
+      secret:
+          secretName: <COOKIE_SECRET_NAME>
+    ```
+    Under `volumeMounts` token, add:
+    ``` 
+    - name: <GITHUB_OAUTH_SECRET_VOLUME_NAME>
+    	  mountPath: <MOUNT_PATH>
```

Comment:
Also since the template is mounting everything in default locations so as to not pass the flags to `deck`, let's use those here too or talk about passing the flags.